server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionComponentsTest.java

```diff
@@ -19,242 +19,597 @@
+
+    wsTester.newRequest()
+      .execute()
+      .assertJson(this.getClass(), "issues_on_different_projects.json");
+  }
+
+  @Test
+  public void do_not_return_module_key_on_single_module_projects() throws IOException {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(defaultOrganization, "P1").setDbKey("PK1"));
+    ComponentDto module = insertComponent(newModuleDto("M1", project).setDbKey("MK1"));
+    ComponentDto file = insertComponent(newFileDto(module, null, "F1").setDbKey("FK1"));
+    RuleDto newRule = newRule();
+    IssueDto issueInModule = IssueTesting.newDto(newRule, file, project).setKee("ISSUE_IN_MODULE");
+    IssueDto issueInRootModule = IssueTesting.newDto(newRule, project, project).setKee("ISSUE_IN_ROOT_MODULE");
+    db.issueDao().insert(session, issueInModule, issueInRootModule);
+    session.commit();
+    indexIssues();
+    indexPermissions();
+
+    SearchWsResponse searchResponse = wsTester.newRequest().executeProtobuf(SearchWsResponse.class);
+    assertThat(searchResponse.getIssuesCount()).isEqualTo(2);
+
+    for (Issues.Issue issue : searchResponse.getIssuesList()) {
+      assertThat(issue.getProject()).isEqualTo("PK1");
+      if (issue.getKey().equals("ISSUE_IN_MODULE")) {
+        assertThat(issue.getSubProject()).isEqualTo("MK1");
+      } else if (issue.getKey().equals("ISSUE_IN_ROOT_MODULE")) {
+        assertThat(issue.hasSubProject()).isFalse();
+      }
+    }
+  }
+
+  @Test
+  public void search_by_project_uuid() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(otherOrganization1, "P1").setDbKey("PK1"));
+    ComponentDto file = insertComponent(newFileDto(project, null, "F1").setDbKey("FK1"));
+    IssueDto issue = IssueTesting.newDto(newRule(), file, project).setKee("82fd47d4-b650-4037-80bc-7b112bd4eac2");
+    db.issueDao().insert(session, issue);
+    session.commit();
+    indexIssues();
+    indexPermissions();
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_PROJECT_UUIDS, project.uuid())
+      .execute()
+      .assertJson(this.getClass(), "search_by_project_uuid.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_PROJECT_UUIDS, "unknown")
+      .execute()
+      .assertJson(this.getClass(), "no_issue.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_COMPONENT_UUIDS, project.uuid())
+      .execute()
+      .assertJson(this.getClass(), "search_by_project_uuid.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_COMPONENT_UUIDS, "unknown")
+      .execute()
+      .assertJson(this.getClass(), "no_issue.json");
+  }
+
+  @Test
+  public void search_since_leak_period_on_project() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(otherOrganization2, "P1").setDbKey("PK1"));
+    ComponentDto file = insertComponent(newFileDto(project, null, "F1").setDbKey("FK1"));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1607)